File: Day 18 : Queues and Stacks/Solution.js
class Solution {
        constructor() {
                this.stack = [];
                this.queue = [];
        }

        pushCharacter = (ch) => {
                this.stack.push(ch);
        };

        enqueueCharacter = (ch) => {
                this.queue.push(ch);
        };

        popCharacter = () => {
                return this.stack.pop();
        };

        dequeueCharacter = () => {
                return this.queue.shift();
        }
}

function main() {
        var s = readLine();
        var len = s.length;
        var obj = new Solution();
        for (var i = 0; i < len; i++) {
                obj.pushCharacter(s.charAt(i));
                obj.enqueueCharacter(s.charAt(i));
        }

        var isPalindrome = true;

        for (var i = 0; i < len / 2; i++) {
                if (obj.popCharacter() != obj.dequeueCharacter()) {
                        isPalindrome = false;
                        break;
                }
        }

        if (isPalindrome)
                console.log("The word, " + s + ", is a palindrome.");
        else
                console.log("The word, " + s + ", is not a palindrome.");
}

main();

File: Day 18 : Queues and Stacks/Solution.rb
class Solution  
  def initialize
    @stack = Array.new
    @queue = Array.new
  end
  
  def push_character(ch)
    @stack.push(ch)
  end
  
  def enqueue_character(ch)
    @queue.push(ch)
  end
  
  def pop_character
    return @stack.pop
  end
  
  def dequeue_character
    return @queue.shift
  end
end

solution = Solution.new
input = gets.chomp
input.split('').each do |c|
  # push the character to stack
  solution.push_character c
  # enqueue the character to queue
  solution.enqueue_character c
end

# To check if input string is palindrome or not
is_palindrome = true

(input.length / 2).times do
  if solution.pop_character != solution.dequeue_character
    is_palindrome = false
    break
  end
end

if is_palindrome
  puts "The word, #{input}, is a palindrome."
else
  puts "The word, #{input}, is not a palindrome."
end
